.github/workflows/testing.yml

```diff
@@ -0,0 +1,63 @@
+name: GitHub Actions Linux Testing
+
+on:
+  push:
+    branches:
+    - master
+    - 'v1.*'
+  pull_request:
+  schedule:
+  - cron: 54 19 * * SUN # weekly at a "random" time
+
+permissions:
+  contents: read
+
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        jre: [8, 11]
+      fail-fast: false # Should swap to true if we grow a large matrix
+
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-java@v2
+      with:
+        java-version: ${{ matrix.jre }}
+        distribution: 'adopt'
+
+    - name: Gradle cache
+      uses: actions/cache@v2
+      with:
+        path: |
+          ~/.gradle/caches
+          ~/.gradle/wrapper
+        key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+        restore-keys: |
+          ${{ runner.os }}-gradle-
+    - name: Maven cache
+      uses: actions/cache@v2
+      with:
+        path: |
+          ~/.m2/repository
+          !~/.m2/repository/io/grpc
+        key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml', 'build.gradle') }}
+        restore-keys: |
+          ${{ runner.os }}-maven-
+    - name: Protobuf cache
+      uses: actions/cache@v2
+      with:
+        path: /tmp/protobuf-cache
+        key: ${{ runner.os }}-maven-${{ hashFiles('buildscripts/make_dependencies.sh') }}
+
+    - name: Build
+      run: buildscripts/kokoro/unix.sh
```

Comment:
Do we still need to set these env vars? https://github.com/grpc/grpc-java/blob/814655cdde5797854289ce5c2ec3e7b80ce0cf44/.travis.yml#L4-L8

Reply:
Yes and no. Yes, we want them, but they are actually managed by unix.sh:
https://github.com/grpc/grpc-java/blob/814655cdde5797854289ce5c2ec3e7b80ce0cf44/buildscripts/kokoro/unix.sh#L41-L46

That said, not having them set globally (and others like skipAndroid) is a bit annoying for the coveralls invocation later and is why I needed to `-x compileJava`.